TODO/designing-design-system-for-complex-products.md

```diff
@@ -1,181 +1,197 @@
+回到 2014 年初，我刚加入 Bypass 时，由于我们的产品非常复杂，出现了大量风格不一致的情况。有不同的按钮、不同的输入框、相同的元素不同的布局、相同的流程不同的交互。我们浪费了很多时间来辩论设计的细节，只是因为我们的产品没有固定的规则。这些不一致也给我们的用户带来了糟糕的体验：他们为此感到困惑，从而难以理解并学习这个产品。
 
-On the development side, because of the variety in code library, it was hard to find the “correct code”. Sometimes developers just wrote new codes instead of reusing the old ones, which led to even more inconsistency. It was a vicious circle, and with our team growing over time, communication and product delivery got harder.
+在开发方面，由于代码库的多样性，总是很难找到「正确的代码」。有时候开发者只是写下新的代码，而不是复用原先的旧代码，从而导致更多的不一致性。这是一个恶性循环，随着时间的推移，我们的团队越来越大，沟通和产品交付也变得越发困难。
 
-### Problems we had ###
+### 我们面临的问题 ###
 
-**1.50 shades of grey:** Since our products were very complex, it was always hard to keep consistency. There were many UI inconsistencies across the system, including different colors, fonts, font sizes, etc.
+**1. 五十度灰:** 由于我们的产品非常复杂，总是难以保持一致。系统中有许多界面风格不一致的地方，包括不同的颜色、字体、字体大小等等。
 
-**2.Lack of rules:** When our designers were building a new feature, it was easy for us to get bogged down with the details. Even simple questions like “Which component should I use?” and “Should I guide users to a new page, or use a pop-up dialog instead?” became very time-consuming to decide on.
+**2. 规则的匮乏:** 当我们的设计师正在构建一个新功能时，我们就很容易深陷细节困境。甚至连一些简单的问题诸如「我应该使用哪个组件？」和「我们应该引导用户进入新的页面，还是弹出对话框呢？」这样的问题都需要花费很多时间来进行决策。
 
-**3.Poor quality of delivery:** Because of the lack of design rules, it was very hard for different teams to communicate with each other. I used to put lots of “red lines” with detailed descriptions on my design, to provide as many details as possible to the developers. But it was not an efficient way to hand off, and sometimes developers did not follow all the details I put on my design.
+**3. 交付质量差:** 由于缺乏设计规则，不同的团队互相沟通是非常困难的。我曾经把很多带有详细描述的「红线」提供给开发者。但这并不是一种有效的手段，有时开发者并没有遵循我所有的设计细节。
 
-Imagine that your team launched a new project and people were celebrating, but you noticed a bunch of design bugs. You must be as happy as everyone else was, no?
+想象一下，你的团队上线了一个新的项目而且其他所有人都在庆祝，但是你却看到了一大堆设计错误。你必须保持和其他人一样开心，是还是不是？
 
 ![](https://cdn-images-1.medium.com/max/800/1*jSQ79eaoZwXK1PfAneOoSw.gif)
 
-The happy you
+> 「开心的你」
 
-**4.Inconsistent code library:** Because there was no rule, developers sometimes just grabbed what they found in the code base and implemented it in new projects. Some other times, they came up with completely new components with a different style, and this further complicated the code library.
+**4. 不一致的代码库:** 由于没有规则，开发者有时只是会复制他们在基础代码库中发现的现有实现直接粘贴到新项目中去。而有时候，他们却会去实现一个风格完全不同的全新组件，这进一步使代码复杂化了。
```

Comment:
“Because there was no rule, developers sometimes just grabbed what they found in the code base and implemented it in new projects. ”
由于没有制定规则，有时候开发者们在一个新项目中只会直接复用基础代码库中已实现的相似代码。